pkg/common/peertracker/npipe_windows.go

```diff
@@ -65,7 +65,7 @@ func isCompareObjectHandlesFound() bool {
 // compareObjectHandles compares two object handles to determine if they
 // refer to the same underlying kernel object
 func compareObjectHandles(firstHandle, secondHandle windows.Handle) error {
-	if isCompareObjectHandlesFound() {
+	if procCompareObjectHandlesErr != nil {
```

Comment:
I couldn't see an easy way to unit test `compareObjectHandles` without refactoring, I just included the one line change in this PR. To make it minimally testable could look something like in the direction of below - changing `compareObjectHandles` into a method on some type so that anything used by it e.g. `syscall.SyscallN` could be overridden on the struct used in testing.

```go
// syscaller wraps anything used by `compareObjectHandles` calls
type syscaller struct {
	procCompareObjectHandlesAddr *uintptr // from procCompareObjectHandles.Addr()
	procCompareObjectHandlesErr  error
	syscallN func(trap uintptr, args ...uintptr) (uintptr, uintptr, syscall.Errno)
}

// windowsSyscaller is the only syscaller that should be used at runtime
var windowsSyscaller syscaller

// init sets up the real syscaller
func init() {
	kernelbase := windows.NewLazyDLL("kernelbase.dll")
	procCompareObjectHandles := kernelbase.NewProc("CompareObjectHandles")
	if err := procCompareObjectHandles.Find(); err != nil {
		windowsSyscaller = syscaller{
			procCompareObjectHandlesErr: err,
			syscallN:                    syscall.SyscallN,
		}
	} else {
		syscallAddr := procCompareObjectHandles.Addr()
		windowsSyscaller = syscaller{
			procCompareObjectHandlesAddr: &syscallAddr,
			syscallN:                     syscall.SyscallN,
		}
	}
}

func (s syscaller) isCompareObjectHandlesFound() bool {
	return s.procCompareObjectHandlesAddr == nil
}

// compareObjectHandles compares two object handles to determine if they
// refer to the same underlying kernel object
func (s syscaller) compareObjectHandles(firstHandle, secondHandle windows.Handle) error {
	if s.procCompareObjectHandlesErr != nil {
		return s.procCompareObjectHandlesErr
	}
	r1, _, e1 := s.syscallN(*s.procCompareObjectHandlesAddr, uintptr(firstHandle), uintptr(secondHandle))
	if r1 == 0 {
		return e1
	}
	return nil
}
```